aten/src/ATen/cpu/vec256/vec256_int.h

```diff
@@ -583,6 +589,18 @@ Vec256<int16_t> inline clamp_min(const Vec256<int16_t>& a, const Vec256<int16_t>
   return _mm256_max_epi16(min_val, a);
 }
 
+Vec256<int32_t> inline convert_8(const int8_t* ptr) {
+  return _mm256_cvtepi8_epi32(_mm_loadl_epi64(reinterpret_cast<const __m128i*>(ptr)));
+}
+
+Vec256<int32_t> inline convert_8(const int32_t* ptr) {
+  return Vec256<int32_t>::loadu(ptr);
+}
+
+Vec256<int32_t> inline convert_8(const uint8_t* ptr) {
+  return _mm256_cvtepu8_epi32(_mm_loadl_epi64(reinterpret_cast<const __m128i*>(ptr)));
+}
```

Comment:
I think a better API would take a size_t template parameter to allow the user to specify how wide they want the conversion. See my discussion about loading full Vec256's of quantized values rather than using `loadl_epi64`. See https://github.com/pytorch/pytorch/blob/b0b0f2c65f4e029e36e17dd8e0f4ebb1c24cb80f/aten/src/ATen/cpu/vec256/vec256_qint.h#L210 where we load a full register and extract and convert subsets of that register

Reply:
Yes, loading might be fine. Key is the signed and unsigned. I could use the functional template specialization to make it better.